src/gameState.ts

```diff
@@ -0,0 +1,299 @@
+import {Position} from "./position"
+import {Probability} from "./probability"
+
+export class GameState{
```

Comment:
...buuuut, I think it even now still has too much inside.

Could you have thought about having a base class to track gamestate and run your AI, while:
* Having a separate class to handle your standard "seek out a new ship" code
* ...and a separate class in the case that you've found a ship, and want to systematically destroy it?